app/lib/widgets/cards/productCard/shared/price.dart

```diff
@@ -0,0 +1,127 @@
+import 'package:flex_ui/utils/extensions.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_annotation/widgetbook_annotation.dart' as widgetbook;
+import 'package:intl/intl.dart';
+
+enum Variant {
+  standard,
+  savings, // Defaults to text decoration linethrough and color 'disabled'
+}
+
+class Price extends StatelessWidget {
+  const Price({
+    super.key,
+    required this.price,
+    this.formatterCallback,
+    this.variant = Variant.standard,
+    this.textStyle,
+  });
+  final double price;
+  final String Function(double)? formatterCallback;
+  final Variant? variant;
+  final TextStyle? textStyle;
+  @override
+  Widget build(BuildContext context) {
+    final defaultTheme = Theme.of(context);
+// @TODO before merge - potentially add theme fallback?
+
+    final formattedPrice = formatterCallback != null
+        ? formatterCallback!(price)
+        : price.toString();
+
+    final Map<Variant, TextStyle> variantTextStyle = {
+      Variant.standard: const TextStyle(),
```

Comment:
Another open question (mostly for @romgrm), do we want to use an optional ThemeExtensions for specific widgets? Flow would be:
1. Look for ThemeExtension (such as FlexPriceTheme)
2. Fallback to Material's default textTheme
3. Optional widget-level override via TextStyle argument

So, in this spot, we'd put..

```
final priceTheme = defaultTheme.extension<FlexPriceTheme>();

PriceVariant.standard: priceTheme?.standardStyle ??
          defaultTheme.textTheme.headlineSmall ??
          const TextStyle(),
```

and then we'd have

```
class FlexPriceThemeData {
  static FlexPriceTheme defaultTheme = FlexPriceTheme(
    standardStyle: const TextStyle(
      fontWeight: FontWeight.bold,
      fontSize: 16,
    ),
)
```

Reply:
If the user use Flex-UI in his own app (with Theme override), I think we don't need to pass it through the widget, as the Text itself will take the new Theme. 
Just the classic TextStyle seems to be enough, no ?